baybe/acquisition.py

```diff
@@ -1,18 +1,43 @@
 """Adapter for making BoTorch's acquisition functions work with other models."""
+from __future__ import annotations
 
+from functools import partial
 from inspect import signature
-from typing import Any, Callable, List, Optional, Type
+from typing import TYPE_CHECKING, Any, Callable, List, Optional, Type
 
 import gpytorch.distributions
 from attr import define
-from botorch.acquisition import AcquisitionFunction
+from botorch.acquisition import (
+    AcquisitionFunction,
+    ExpectedImprovement,
+    PosteriorMean,
+    ProbabilityOfImprovement,
+    UpperConfidenceBound,
+    qExpectedImprovement,
+    qProbabilityOfImprovement,
+    qUpperConfidenceBound,
+)
 from botorch.models.gpytorch import Model
 from botorch.posteriors import Posterior
 from botorch.posteriors.gpytorch import GPyTorchPosterior
-from torch import Tensor, cat, squeeze
 
 from baybe.surrogates.base import Surrogate
 
+if TYPE_CHECKING:
+    from torch import Tensor
+
+mapping = {
```

Comment:
since this is now not a function-local variable anymore Id give it a more precise name, ie mapping of what?

Reply:
see other comment